packages/react-server/src/ReactFizzNewContext.js

```diff
@@ -244,7 +245,14 @@ export function popProvider<T>(context: ReactContext<T>): ContextSnapshot {
     }
   }
   if (isPrimaryRenderer) {
-    prevSnapshot.context._currentValue = prevSnapshot.parentValue;
+    const value = prevSnapshot.parentValue;
+    if (value === REACT_SERVER_CONTEXT_DEFAULT_VALUE_NOT_LOADED) {
+      prevSnapshot.context._currentValue =
+        // $FlowExpectedError - Effectively refined context to ServerContext
+        (prevSnapshot.context: ReactServerContext<any>)._defaultValue;
```

Comment:
You don't have to refine this since `_defaultValue` is on client contexts too. It's not necessary but doesn't hurt to have it on client contexts too.